python/metatensor-learn/metatensor/learn/data/collate.py

```diff
@@ -0,0 +1,112 @@
+"""
+Module containing a collate function for use in a Dataloader.
+"""
+from collections import namedtuple
+from typing import List, Tuple
+
+import torch
+import metatensor
+from metatensor import TensorMap
+
+
+def collate_fn(batch: List[Tuple]) -> Tuple:
```

Comment:
As I understand the changes here, we don't support mixed data types in the dataset, right? 

This would not be able to handle something like `Dataset(frames=[system, system], energy=[TensorMap(), TensorMap()])`? I think we want to have something like this.

Reply:
No, this is supported! You can pass any arbitrary object in each of the fields